### PR TITLE
Add light style to box

### PR DESCRIPTION
**Pull request by hcwong**
Using Markbind's PR template cos there isn't one for vue-strap

**What is the purpose of this pull request? (put "X" next to an item, remove the rest)**

• [x] Enhancement to an existing feature

Fixes https://github.com/MarkBind/markbind/issues/909

**What is the rationale for this request?**
Requested enhancement

**What changes did you make? (Give an overview)**
I added an additional prop to box called `light` that allows the user to make the tipbox lighter by about 20%

**Provide some example code that this change will affect:**

```js
// Current
<box type="success">
    success
</box>

// Light
<box type="success" light>
    success light
</box>
```

**Is there anything you'd like reviewers to focus on?**
I am terrible at colors and would appreciate any ideas on the color scheme I chose.

Also, I do not think that the colors should be defined in the `Tipbox.vue` file itself as colors should be consistent across the repo. Any suggestions on where I should define the light color constants?

**Testing instructions:**
Same as DG of Markbind.

**Proposed commit message: (wrap lines at 72 characters)**
Adds a light style to TipBox via an additional prop

**Screenshots**
**Before (19.5%)**
Screen Shot 2020-01-03 at 10 52 11 PM

**After change (36%)**
Screen Shot 2020-01-07 at 5 01 38 PM

### CODE REVIEW

**Comment by yamgent** [comment]
Apologies, I missed this PR because GitHub doesn't notify if only the title of the PR changes. You might want to make a "Ready for Review" comment when your PR is ready in order to notify other contributors in the future.

> Also, I do not think that the colors should be defined in the `Tipbox.vue` file itself as colors should be consistent across the repo. Any suggestions on where I should define the light color constants?

You can create a .css file inside `src/utils/`. Seems like Vue allows importing of css:  https://stackoverflow.com/a/43862129

> I am terrible at colors and would appreciate any ideas on the color scheme I chose.

I think it is quite hard to differentiate between the light colors and the normal colors. Can it be lighter?

**Comment by damithc** [comment]
* w.r.t. screenshot, I don't think `danger` is a valid type. Should be `error`?

* Yes, hard to see the difference between the normal and the light. 

* For the light mode, should we add left boarder by default (as the background might be too light to set the box apart from the rest), like this? 
image
Just a thought.

File: src/TipBox.vue [hunk 1]
```diff
@@ -76,6 +81,26 @@
             return 'alert-default'
         }
       },
+      lightStyle() {
+        if (this.light) {
+            switch(this.type) {
```
Comment (yamgent) [inline src/TipBox.vue]: Space between `switch` and `(`

File: src/TipBox.vue [hunk 2]
```diff
@@ -76,6 +81,26 @@
...
+            case 'tip':
+                return 'alert-success-light alert-border-left';
+            case 'important':
+            case 'wrong':
+                return 'alert-danger-light alert-border-left';
+            default:
+                return '';
+            }
+        }
+        return "";
```
Comment (yamgent) [inline src/TipBox.vue]: Should use `''` instead of `""`.

File: src/utils/colors.css [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+}
+
+.alert-danger-light {
+    background-color: #fae5e7;
+    border-color: #f8dadd;
+}
+
+.alert-border-left {
+    border-left: 5px solid;
+}
```
Comment (yamgent) [inline src/utils/colors.css]: Missing ending newline for this file.

**Comment by yamgent** [comment]
Also you can now squash your commits.

**Comment by damithc** [comment]
Good work. I would like to see how these options would look like, if possible.
* An even lighter shade of background
* Use regular font color for text instead of using different colors based on the box type

**Comment by hcwong** [comment]
> Good work. I would like to see how these options would look like, if possible.
> 
> * An even lighter shade of background
> * Use regular font color for text instead of using different colors based on the box type

Hi Prof, actually what do you mean by regular font color? the font color currently being used now is what bootstrap provides by default

**Comment by damithc** [comment]
> Hi Prof, actually what do you mean by regular font color? the font color currently being used now is what bootstrap provides by default

I meant regular black instead of green, red, etc. e.g.,
image

Just want to see how it looks.

**Comment by hcwong** [comment]
> > Hi Prof, actually what do you mean by regular font color? the font color currently being used now is what bootstrap provides by default
> 
> I meant regular black instead of green, red, etc. e.g.,
> image
> 
> Just want to see how it looks.

**(46% lightening)**
Screen Shot 2020-01-08 at 3 13 01 PM

**black color**
*Looks a bit weird IMO*

Screen Shot 2020-01-08 at 3 14 35 PM

**Comment by damithc** [comment]
I was looking for something like this:
image

Is something like that possible?

I got the above using current syntax
```html
<box background-color="#e6ffe6" border-left-color="#009900">

<span class="text-success">**:fas-lightbulb:**</span> **This is a tip**. Tip description.
</box>

<box background-color="#ffe6e6" border-left-color="#cc0000">

<span class="text-danger">**:fas-exclamation:**</span> **This is a warning**. Warning description.
</box>
```

**Comment by hcwong** [comment]
> I was looking for something like this:
> image
> 
> Is something like that possible?
> 
> I got the above using current syntax
> 
> ```
> <box background-color="#e6ffe6" border-left-color="#009900">
> 
> <span class="text-success">**:fas-lightbulb:**</span> **This is a tip**. Tip description.
> </box>
> 
> <box background-color="#ffe6e6" border-left-color="#cc0000">
> 
> <span class="text-danger">**:fas-exclamation:**</span> **This is a warning**. Warning description.
> </box>
> ```

Hi Prof, I can add it an option to make text black/some other color, if i get you correctly

**Comment by damithc** [comment]
> Hi Prof, I can add it an option to make text black/some other color, if i get you correctly

Can you make the `light` option produce exactly this combination? i.e., icon and left-border uses the dark shade of the style, no other borders, and the text is black.

 image

In fact, I'm thinking of not having the background color at all. My reason for asking for the light style is that the current box style has too much visual weight which makes me reluctant to use them as they disrupt the visual flow of the page too much. I think it is the background color that cause the visual 'heaviness'. See example below (the second box is lighter in visual weight and blends in better with the page):
image

Perhaps we can configure the `light` mode to be like the 2nd box above (i.e., left edge and icon follow the style color but the background is light grey and the text is black)? Sorry to ask for many variations; as we have a `box` already, this additional option should provide sufficient value. Making the shade lighter alone doesn't seem to make much of a difference.

**Comment by hcwong** [comment]
> In fact, I'm thinking of not having the background color at all. My reason for asking for the light style is that the current box style has too much visual weight which makes me reluctant to use them as they disrupt the visual flow of the page too much. I think it is the background color that cause the visual 'heaviness'. See example below (the second box is lighter in visual weight and blends in better with the page):
> image
> 
> Perhaps we can configure the `light` mode to be like the 2nd box above (i.e., left edge and icon follow the style color but the background is light grey and the text is black)? Sorry to ask for many variations; as we have a `box` already, this additional option should provide sufficient value. Making the shade lighter alone doesn't seem to make much of a difference.

Hi Prof, sflr was sick. I made 3 options below.

**1. Light mode, dark border, black text, no border**
Screen Shot 2020-01-10 at 3 51 29 PM

**2. Light grey bg, dark border, black text, no border**
Screen Shot 2020-01-10 at 3 46 26 PM

**3. Light grey bg, light border, black text, no border**
Screen Shot 2020-01-10 at 3 46 44 PM


Which one is preferable?

**Comment by damithc** [comment]
> Which one is preferable?

These default colors are not very nice. For example, this one is almost black/brown.
  image

Is it possible to use the regular bootstrap colors palette (used in panels)? This this:
image

**Comment by hcwong** [comment]
> Is it possible to use the regular bootstrap colors palette (used in panels)? This this:
> image

Yep it is possible. I'm using the Panels color palette here:
Screen Shot 2020-01-10 at 6 06 55 PM

**Comment by hcwong** [comment]
> > Which one is preferable?
> 
> These default colors are not very nice. For example, this one is almost black/brown.
> image
> 
> Is it possible to use the regular bootstrap colors palette (used in panels)? This this:
> image

Hi Prof yes it is possible. I'm using the color palette of panels here:
Screen Shot 2020-01-10 at 6 06 55 PM

**Comment by damithc** [comment]
> Hi Prof yes it is possible. I'm using the color palette of panels here:
> Screen Shot 2020-01-10 at 6 06 55 PM

Nice. I'm happy with this version. :-)

**Comment by damithc** [comment]
This feature in action https://nus-cs2113-ay1920s2.github.io/website/schedule/week4/tutorial.html
Good work @hcwong Thanks to reviewers too.

**Pull request merged**